src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -164,13 +165,33 @@ private Task StartAsyncCore()
                         _logger.HttpConnectionClosed();
                         return;
                     }
-
-                    _transport = _transportFactory.CreateTransport(GetAvailableServerTransports(negotiationResponse));
+                    _serverTransports = GetAvailableServerTransports(negotiationResponse);
+                    _transport = _transportFactory.CreateTransport(_serverTransports);
                     connectUrl = CreateConnectUrl(Url, negotiationResponse);
                 }
 
                 _logger.StartingTransport(_transport, connectUrl);
-                await StartTransport(connectUrl);
+                var validTransports = _requestedTransportType & _serverTransports;
+                while (validTransports > 0)
+                {
+                    try
+                    {
+                        await StartTransport(connectUrl);
```

Comment:
We should pass the transport we want to try in here directly rather than assuming it will pick the next applicable transport.